src/ppx/scene/scene_scene.cpp

```diff
@@ -122,5 +163,50 @@ ppx::Result Scene::AddNode(scene::NodeRef&& node)
     return ppx::SUCCESS;
 }
 
+scene::ResourceIndexMap<scene::Sampler> Scene::GetSamplersArrayIndexMap() const
+{
+    auto array = mResourceManager->GetSamplers();
+
+    std::unordered_map<const scene::Sampler*, uint32_t> indexMap;
+    //
+    const uint32_t objectCount = CountU32(array);
+    for (uint32_t idx = 0; idx < objectCount; ++idx) {
```

Comment:
Since the array form is not useful once we have this map, shall you change the map creation to:

```cpp
size_t index = 0;
std::unordered_map<const scene::Sampler*, uint32_t> indexMap;
for (const auto& item : mSamplers()) {
     indexMap[item] = index;
     ++index;
}
```

?

Reply:
It could, but the reason why there's all the annoying `blahCount` stuff is for debugging should a breakpoint need to be placed in there. In case we have an IDE that isn't awesome at letting us view STL containers.

Reply:
Sorry, I misread what you said originally and my brain processed while I was driving. Good suggestion. I've made the changes.